integrationTest/integration_test.go

```diff
@@ -290,11 +293,13 @@ var tests = []*TestCase{
 	tc("Change it", cname("foo", "google2.com.")),
 	tc("Change to A record", a("foo", "1.2.3.4")),
 	tc("Change back to CNAME", cname("foo", "google.com.")),
+	tc("Record pointing to @", cname("foo", "**current-domain**")),
```

Comment:
I managed to create tests for this by using special value for test cases and rewriting the value to `domainName` on `runTests`. Not very pretty but works.